src/DGmethods/DGmodel.jl

```diff
@@ -1,12 +1,20 @@
-struct DGModel{BL,G,NFND,NFD,GNF}
+struct DGModel{BL,G,NFND,NFD,GNF,AS,DS}
   balancelaw::BL
   grid::G
   numfluxnondiff::NFND
   numfluxdiff::NFD
   gradnumflux::GNF
+  auxstate::AS
+  diffstate::DS
+end
+function DGModel(balancelaw, grid, numfluxnondiff, numfluxdiff, gradnumflux;
+                 auxstate=create_auxstate(balancelaw, grid),
+                 diffstate=create_diffstate(balancelaw, grid))
+  DGModel(balancelaw, grid, numfluxnondiff, numfluxdiff, gradnumflux, auxstate,
+          diffstate)
 end
 
-function (dg::DGModel)(dQdt, Q, param, t; increment=false)
+function (dg::DGModel)(dQdt, Q, t; increment=false)
```

Comment:
I'm happy to move the functionality, but it might make sense to keep an empty argument here (i.e. `(dg::DGModel)(dQdt, Q, _, t; increment=false)`) for the purposes of compatibility with DifferentialEquations.jl.

Reply:
Is this what you were thinking: https://github.com/climate-machine/CLIMA/pull/452/commits/f5a9548b9da85da797f58cf376e58039b13ee90a

Reply:
That works. I guess the question is whether you would want to change the interface for the ODE solvers as well, but since that's probably not a pressing issue at the moment.

Reply:
I think that e1fcee5 is what you suggest, but now I get an error
```
all-underscore identifier used as rvalue around /Users/jekozdon/codes/CLIMA/src/DGmethods/DGmodel.jl:18
```
if I use the underscores as opposed to catching the `p` explicitly. Not sure why. Spent a little time trying to track it down, but no luck.